politeiad/backendv2/tstorebe/plugins/pi/cmds.go

```diff
@@ -29,6 +30,126 @@ const (
 	dataDescriptorBillingStatus = pluginID + "-billingstatus-v1"
 )
 
+// cmdSummary returns the pi summary of a proposal.
+func (p *piPlugin) cmdSummary(token []byte) (string, error) {
+	proposalStatus, err := p.proposalStatus(token)
+	if err != nil {
+		return "", err
+	}
+
+	// Prepare reply
+	sr := pi.SummaryReply{
+		Summary: pi.ProposalSummary{
+			Status: proposalStatus,
+		},
+	}
+	reply, err := json.Marshal(sr)
+	if err != nil {
+		return "", err
+	}
+
+	return string(reply), nil
+}
+
+// proposalStatusApproved returns the proposal status of an apprvoed proposal.
+func (p *piPlugin) proposalStatusApproved(token []byte) (pi.PropStatusT, error) {
+	// Get billing status to determine the proposal status.
+	bsc, err := p.billingStatusChange(token)
+	if err != nil {
+		return pi.PropStatusUnvetted, err
+	}
+	// If a billing status of an approved proposal not set then the
+	// proposal is considered as active.
+	if bsc == nil {
+		return pi.PropStatusActive, nil
+	}
+	switch bsc.Status {
+	case pi.BillingStatusClosed:
+		return pi.PropStatusClosed, nil
+	case pi.BillingStatusCompleted:
+		return pi.PropStatusCompleted, nil
+	}
+	// Shouldn't happen return an error
+	return pi.PropStatusUnvetted,
```

Comment:
Invalid statuses should be the zero value of the type.  For ints we use 0.  For strings it should be an empty string.

Now that I think about it, we should make this explicit in the plugin API.  Please add a `PropStatusInvalid PropStatusT = ""` to the plugin API and return this invalid status for failure cases like this.